tools/squidclient/squidclient.cc

```diff
@@ -510,10 +508,22 @@ main(int argc, char *argv[])
         if (max_forwards > -1) {
             msg << "Max-Forwards: " << max_forwards << "\r\n";
         }
-        if (ProxyAuthorization.user)
-            ProxyAuthorization.commit(msg);
-        if (OriginAuthorization.user)
-            OriginAuthorization.commit(msg);
+        if (ProxyAuthorization.user) {
+            ProxyAuthorization.commit(msg); // handles missing passwords
+        } else if (ProxyAuthorization.password) {
+            std::cerr << "ERROR: Proxy authentication username is missing\n";
```

Comment:
Please move this enforcement up/earlier, ASAP after all the named command-line options are parsed by the `while` loop. We should not make these basic configuration  checks when processing each individual URL. A misconfigured squidclient should quit on startup instead. Fixing this will also simplify this convoluted code logic a bit.

Same for OriginAuthorization, of course.

Reply:
Done in [PR1251](https://github.com/squid-cache/squid/pull/1251/).